src/main/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplication.scala

```diff
@@ -343,6 +347,7 @@ private[spark] class ArmadaClientApplication extends SparkApplication {
     // Note: Don't filter empty strings here - validation should handle that later
     val queue          = conf.get(ARMADA_JOB_QUEUE)
     val jobSetId       = conf.get(ARMADA_JOB_SET_ID)
+    val runAsUser      = conf.get(ARMADA_RUN_AS_USER)
```

Comment:
I think it is better to specify the run as user via the [SecurityContext](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod) as this is a native Kubernetes feature than to specifically set the `runAsUser` field.